lightningd/plugin.c

```diff
@@ -631,11 +631,12 @@ static bool plugin_rpcmethod_add(struct plugin *plugin,
 				 const char *buffer,
 				 const jsmntok_t *meth)
 {
-	const jsmntok_t *nametok, *desctok, *longdesctok, *usagetok;
+	const jsmntok_t *nametok, *categorytok, *desctok, *longdesctok, *usagetok;
 	struct json_command *cmd;
 	const char *usage;
 
 	nametok = json_get_member(buffer, meth, "name");
+	categorytok = json_get_member(buffer, meth, "category");
```

Comment:
I'm not sure we actually want to allow plugins to change their category at all. It would be nice if we could group their RPC methods by the plugin that exposes them, but that doesn't play nicely with the numeric category used in the rest of the PR.

Then again, having `pay` and `autoclean` be categorized in plugins seems weird as well.

Reply:
> I'm not sure we actually want to allow plugins to change their category at all. It would be nice if we could group their RPC methods by the plugin that exposes them, but that doesn't play nicely with the numeric category used in the rest of the PR.

Yes, I did the code for Pylightning similarly to what I added to libplugin.c and did not push it because I remarked the same. Maybe we can assume that C plugins are internal and that others external so leave the possibility to chose the category for internal plugins in libplugin.c and not in Pylightning ? Then we can leave this PR as a base with unsorted plugin commands and i'll add another one that pass from : 
```
=== PLUGIN ===
"command a" from plugin 1

"command b" from plugin 2

"command c" from plugin 2
```
to
```
=== PLUGIN ===
plugin 1
    command a

plugin 2
    command b
    command c
```

Reply:
> Maybe we can assume that C plugins are internal and that others external so leave the possibility to chose the category for internal plugins in libplugin.c and not in Pylightning ?

The language a plugin is written in should not matter -- we may someday want to add a sufficiently-useful plugin to the "stock" C-lightning release, and if it be written in Brainfuck, as long as it is useful, it should be acceptable.

Perhaps category should be a string and the provenance of the command (internal or plugin) be an enumerated field.

Reply:
I don't say that the language matters but that for now internal plugins are written with libcplugin and that I think it's ok to let the possibility of specifying the command category in it (and only this one).